src/utils/fields.ts

```diff
@@ -23,6 +23,39 @@ export function extractFields(data: DataFrame) {
   return result;
 }
 
+type labelName = string;
+type labelValue = string;
+
+export function getCardinalityMapFromFrame(frame: DataFrame) {
```

Comment:
copy/paste from the table UI code, so it's doing some stuff right now that we don't need, like returning the length of the longest string for that field.

There is a performance cost with doing this, typically <3ms, and the table will need this map as well, so ideally we'd be able to expose the result of the calculation to the table

Reply:
Cleaned it up, still seeing <3ms with worst case data, but it will save more then it uses by preventing more calls to the datasource for single cardinality time series panels, besides avgFields like 'duration', 'count', 'total', and 'bytes', which are included to allow time range selection.